google/oauth2/service_account.py

```diff
@@ -448,6 +487,14 @@ def refresh(self, request):
             self.token = access_token
             self.expiry = expiry
 
+        # In case of trust boundary never been fetched.
```

Comment:
Can this be merged with the above section? We should only fetch the trust boundary in one section of code.

Reply:
The difference is in this case, the token is a new token and the case above is using a valid token.